UM/Math/Polygon.py

```diff
@@ -141,6 +141,28 @@ def intersectionConvexHulls(self, other: "Polygon") -> "Polygon":
 
         return Polygon(points = [list(p) for p in polygon_intersection.exterior.coords[:4]])
 
+    #  Computes the convex hull of the union of the convex hulls of this and another polygon.
+    #
+    #   \param other The other polygon to combine convex hulls with.
+    #   \return The convex hull of the union of the two polygons' convex hulls.
+    def unionConvexHulls(self, other: "Polygon") -> "Polygon":
+        my_hull = self.getConvexHull()
+        other_hull = other.getConvexHull()
+
+        if len(my_hull._points) <= 2:
```

Comment:
We should probably change the isValid function to do this check, since any polygon with <=2 points is not valid.